master/internal/rm/kubernetes/pods.go

```diff
@@ -251,6 +302,10 @@ func (p *pods) startClientSet(ctx *actor.Context) error {
 }
 
 func (p *pods) getMasterIPAndPort(ctx *actor.Context) error {
```

Comment:
random: introspecting the service pointing at us is kind of an interesting way to do this in the first place. in a past life i would've terraformed a service and deployment, and dropped some outputs of the service tf into the deployment. this works, just isn't very flexible/it deployment-specific.